internal/mempool/reactor.go

```diff
@@ -53,12 +53,12 @@ type Reactor struct {
 	// goroutines.
 	peerWG sync.WaitGroup
 
+	mtx         tmsync.Mutex
+	peerClosers map[types.NodeID]context.CancelFunc
```

Comment:
An option for your consideration: For locks that protect only one thing, I sometimes find this idiom useful:

```go
peerClosers struct {
   tmsync.Mutex
   m map[types.NodeID]context.CancelFunc
}

// usage:
r.peerClosers.Lock()
if c, ok := r.peerClosers.m[id]; ok { c() }
r.peerClosers.Unlock()
```

Not necessary, but makes the scope of the mutex clear. YMMV.

Reply:
yeah I've done this in the past, and like it as an idiom. 

this is just moving code around, I initially thought i could delete it, but it makes sense to leave it for now.